torch_geometric/nn/aggr/patch_transformer.py

```diff
@@ -0,0 +1,159 @@
+    ):
+        super().__init__()
+        self.channels = channels
+        self.time_dim = time_dim
+        self.hidden_dim = hidden_dim
+        self.out_dim = out_dim
+        self.heads = heads
+        self.layer_norm = layer_norm
+        self.patch_size = patch_size
+        self.dropout = dropout
+        self.max_edge = max_edge
+        self.aggr = aggr
+        if (self.aggr is None):
+            self.aggr = ["mean"]
+
+        # encoders and linear layers
+        self.feat_encoder = FeatEncoder(self.channels, self.hidden_dim,
+                                        time_dim=self.time_dim)
+        self.layernorm = torch.nn.LayerNorm(self.hidden_dim)
+        self.atten_blocks = torch.nn.ModuleList([
+            TransformerBlock(dims=self.hidden_dim, heads=self.heads,
+                             dropout=self.dropout)
+            for _ in range(num_encoder_blocks)
+        ])
+
+        # input is node feature and transformer node embed
+        self.mlp_head = torch.nn.Linear(self.hidden_dim * (len(self.aggr)),
+                                        self.out_dim)
+
+        # padding
+        self.stride = self.patch_size
+        self.pad_projector = torch.nn.Linear(self.patch_size * self.hidden_dim,
+                                             self.hidden_dim)
+        self.pe = PositionalEncoding1D(self.hidden_dim)
+
+    def reset_parameters(self):
+        self.feat_encoder.reset_parameters()
+        for atten_block in self.atten_blocks:
+            atten_block.reset_parameters()
+        self.mlp_head.reset_parameters()
+        self.pad_projector.reset_parameters()
+
+    @disable_dynamic_shapes(required_args=['dim_size', 'max_num_elements'])
+    def forward(
+        self,
+        x: Tensor,
+        index: Tensor,
+        ptr: Optional[Tensor] = None,
+        dim_size: Optional[int] = None,
+        dim: int = -2,
+        max_num_elements: Optional[int] = None,
+    ) -> Tensor:
+
+        # x = [num_edges, input_channels]
+        x = self.feat_encoder(x)
+
+        # x = [num_edges, hidden_dim]
+        edge_feat_size = x.shape[1]
+
+        # x = [batch_size, max_num_elements, -1]
+        x, mask = self.to_dense_batch(x, index, ptr, dim_size, dim,
+                                      max_num_elements=max_num_elements)
+
+        x_feat = x.view(mask.shape[0] * max_num_elements, edge_feat_size)
+        # x_feat = [batch_size, # patch, patch_size * edge_feat_size]
+        x_feat = x_feat.view(-1, self.max_edge // self.patch_size,
+                             self.patch_size * x_feat.shape[-1])
+        # x_feat = [batch_size, max_edge, edge_feat_size]
+        x_feat = self.pad_projector(x_feat)
+
+        # x_feat = [batch_size, max_edge, edge_feat_size]
+        x_feat = x_feat + self.pe(x_feat)
+
+        # x_feat = [batch_size, max_edge, edge_feat_size]
+        for atten_block in self.atten_blocks:
+            x_feat = atten_block(x_feat)
+
+        x_feat = self.layernorm(x_feat)
```

Comment:
<img width="724" alt="image" src="https://github.com/user-attachments/assets/c2948823-786e-4c27-ac8d-e35820e7586d">
Do we need this `layernorm`, in the paper seems they just have pooling and FC

Reply:
This was in the original implementation as well and seem to produce slightly better performance, let's keep it for now